### DISCUSSION TITLE
Multigrid Solvers in Moose

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#17338] [Q&A Modules: General] [answered]
Does Moose have the ability to solve Finite Element Problems using multi-grid solvers?

**Comment by aeslaughter** [comment]
@fdkong could you elaborate on our capabilities.

**Comment by fdkong** [comment, marked as answer]
Yes, there are several MG solvers available in MOOSE. For Algebraic MG, we can use hypre (-pc_type hypre), ML (-pc_type ml), and GAMG (-pc_type gamg).

 For geometric MG, we have something in libmesh, but we do not explore much yet.
 
 What is your context? What are you looking for?

**Reply by TLWise** [reply]
For the combined solution of the equation of motion (F=ma) and the navier stokes, I am seeking to solve each equation in two separate domains then apply the solution to a third domain which is then applied as an initial condition on the navier stokes domain for successive iterations in time.  Here are some additional details that I seek advisement on for implementation in moose. An example of the method applied in moose would be an even greater starting point for my research: 

Employ a (rough) boundary parametrization which sufficiently describes all large-scale structures with regard to the boundary conditions.

Treat all fine- scale features as interior objects such that the corresponding components in all matrices and vectors are unknown degrees offreedom which are implicitely incorporated into all iterative solution steps. 

Utilize filtering techniques to project the corresponding vector components onto the subspace of "correct" boundary conditions, before and directly after each iterative substep.

**Reply by fdkong** [reply]
As far as I know, there is no example close to what you want. I might not completely understand your description here.


You may explore a multi-app system that allows to solve different physics and transfer solutions from one system to another. After transfer, you can use the transferred solution as the initial guess for the current system. The projection step can be done as part of the transfer, but you might need to implement your customized projection transfer if the existing transfers do not work for you.

**Reply by TLWise** [reply]
Thank you very much for this information. I think that it will provide a good start in the right direction. I will try the multi app approach.
